src/doc/src/reference/unstable.md

```diff
@@ -0,0 +1,188 @@
+
+Just include the `package` key to specify the actual name of the dependency.
+You must include `cargo-features` at the top of your `Cargo.toml`.
+
+```toml
+cargo-features = ["rename-dependency"]
+
+[package]
+name = "mypackage"
+version = "0.0.1"
+
+[dependencies]
+foo = "0.1"
+bar = { version = "0.1", registry = "custom", package = "foo" }
+baz = { git = "https://github.com/example/project", package = "foo" }
+```
+
+In this example, three crates are now available in your Rust code:
+
+```rust
+extern crate foo;  // crates.io
+extern crate bar;  // registry `custom`
+extern crate baz;  // git repository
+```
+
+
+### publish-lockfile
+* Original Issue: [#2263](https://github.com/rust-lang/cargo/issues/2263)
+* PR: [#5093](https://github.com/rust-lang/cargo/pull/5093)
+
+When creating a `.crate` file for distribution, Cargo has historically
+not included the `Cargo.lock` file.  This can cause problems with
+using `cargo install` with a binary.  You can specify that your package
+should include the `Cargo.lock` file when using `cargo package` or `cargo publish`
+by specifying the `publish-lockfile` key in `Cargo.toml`.  This also requires the
+appropriate `cargo-features`:
+
+```toml
+cargo-features = ["publish-lockfile"]
+
+[project]
+...
+publish-lockfile = true
+```
+
+
+### Offline Mode
+* Original Issue: [#4686](https://github.com/rust-lang/cargo/issues/4686)
+
+The `-Z offline` flag prevents Cargo from attempting to access the network for
+any reason.  Typically Cargo will stop with an error if it wants to access the
+network and it is not available.
+
+Beware that this may result in different dependency resolution than online
+mode.  Cargo will restrict itself to crates that are available locally, even
+if there might be a newer version as indicated in the local copy of the index.
+
+### no-index-update
+* Original Issue: [#3479](https://github.com/rust-lang/cargo/issues/3479)
+
+The `-Z no-index-update` flag ensures that Cargo does not attempt to update
+the registry index.  This is intended for tools such as Crater that issue many
+Cargo commands, and you want to avoid the network latency for updating the
+index each time.
+
+### avoid-dev-deps
+* Original Issue: [#4988](https://github.com/rust-lang/cargo/issues/4988)
+* Stabilization Issue: [#5133](https://github.com/rust-lang/cargo/issues/5133)
+
+When running commands such as `cargo install` or `cargo build`, Cargo
+currently requires dev-dependencies to be downloaded, even if they are not
+used.  The `-Z avoid-dev-deps` flag allows Cargo to avoid downloading
+dev-dependencies if they are not needed.  The `Cargo.lock` file will not be
+generated if dev-dependencies are skipped.
+
+### minimal-versions
+* Original Issue: [#4100](https://github.com/rust-lang/cargo/issues/4100)
+
+When a `Cargo.lock` file is generated, the `-Z minimal-versions` flag will
+resolve the dependencies to the minimum semver version that will satisfy the
+requirements (instead of the greatest version).
```

Comment:
So, the purpose of `--minimal-versions` is to **check** that minimal version requirements in your Cargo.toml are correct. So, it's sole intended usage is during CI, you never need to use `--minimal-versions` in real life. Perhaps we should even rename the flag to make this more obvious? 

What errors does this check for? Suppose, you are building a library, and have a `foo = "1.0.0"` dependency in your Cargo.toml. Because you are developing a library, you don't commit `Cargo.lock`. Now, some time passes, the author of `foo` releases `foo 1.1.0` and it includes some new features. Because you haven't committed `Cargo.lock`, there's a high chance that you'll be compiling your library with `foo 1.1.0` during development. And that means that you can accidentally use the features from `1.1.0`, although your Cargo.toml says 1.0.0!

It seems that such overly-optimistic lower bonds are pretty common in the ecosystem: because Cargo always picks the latest versions, it does not actually matter in practice. However, it might be slightly better to be aware of what your minimal requirements actually are, and `--minimal-versions` flag help with this. 

So, let's add something like this to docs:

"the intended use-case of the flag is to check, during continuous integration, that the versions, specified in Cargo.toml are correct. That is, that if Cargo.toml says `foo = "1.0.0"`, you don't accidentally depend on features, added only in `foo 1.5.0`".

Reply:
Ah, that makes sense!  I'm not sure why it didn't click with me earlier.  I don't think it needs to be renamed.